test/unit/test_natural_language_understanding_v1.py

```diff
@@ -0,0 +1,376 @@
+    #--------------------------------------------------------
+    @responses.activate
+    def test_analyze_required_response(self):
+        # Check response with required params
+        body = self.construct_required_body()
+        response = fake_response_AnalysisResults_json
+        send_request(self, body, response)
+        assert len(responses.calls) == 1
+
+    #--------------------------------------------------------
+    # Test 3: Send empty data and check response
+    #--------------------------------------------------------
+    @responses.activate
+    def test_analyze_empty(self):
+        check_empty_required_params(self, fake_response_AnalysisResults_json)
+        check_missing_required_params(self)
+        assert len(responses.calls) == 0
+
+    #-----------
+    #- Helpers -
+    #-----------
+    def make_url(self, body):
+        endpoint = '/v1/analyze'
+        url = '{0}{1}'.format(base_url, endpoint)
+        return url
+
+    def add_mock_response(self, url, response):
+        responses.add(responses.POST,
+                      url,
+                      body=json.dumps(response),
+                      status=200,
+                      content_type='application/json')
+
+    def call_service(self, body):
+        service = NaturalLanguageUnderstandingV1(
+            authenticator=NoAuthAuthenticator(),
+            version='2019-07-12',
+        )
+        service.set_service_url(base_url)
+        output = service.analyze(**body)
+        return output
+
+    def construct_full_body(self):
+        body = dict()
+        body.update({
+            "features":
+                Features._from_dict(
+                    json.loads(
+                        """{"concepts": {"limit": 5}, "emotion": {"document": true, "targets": []}, "entities": {"limit": 5, "mentions": true, "model": "fake_model", "sentiment": false, "emotion": false}, "keywords": {"limit": 5, "sentiment": false, "emotion": false}, "metadata": {}, "relations": {"model": "fake_model"}, "semantic_roles": {"limit": 5, "keywords": true, "entities": true}, "sentiment": {"document": true, "targets": []}, "categories": {"explanation": false, "limit": 5}, "syntax": {"tokens": {"lemma": false, "part_of_speech": true}, "sentences": false}}"""
+                    )),
+            "text":
+                "string1",
+            "html":
+                "string1",
+            "url":
+                "string1",
+            "clean":
+                True,
+            "xpath":
+                "string1",
+            "fallback_to_raw":
+                True,
+            "return_analyzed_text":
+                True,
+            "language":
+                "string1",
+            "limit_text_characters":
+                12345,
+        })
+        return body
+
+    def construct_required_body(self):
+        body = dict()
+        body.update({
+            "features":
+                Features._from_dict(
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['Features' may be undefined, or defined from star imports: ibm_watson.natural_language_understanding_v1](https://app.codacy.com/manual/watson-developer-cloud/python-sdk/pullRequest?prid=4835642)